pilot/pkg/model/validation.go

```diff
@@ -1919,16 +1933,34 @@ func validateDestination(destination *routingv2.Destination) (errs error) {
 		return
 	}
 
-	// TODO: validate short name / FQDN
-	if destination.Name == "" {
-		errs = appendErrors(errs, fmt.Errorf("route rule destination should not be empty"))
+	errs = appendErrors(errs, validateHost(destination.Name))
+	if destination.Subset != "" {
+		errs = appendErrors(errs, validateSubsetName(destination.Subset))
+	}
+	if destination.Port != nil {
+		errs = appendErrors(errs, validatePortSelector(destination.Port))
 	}
 
-	// TODO: Subset
+	return
+}
 
-	// TODO: Port
+func validateSubsetName(name string) error {
+	if !IsDNS1123Label(name) {
+		return fmt.Errorf("subnet name is invalid: %s", name)
+	}
+	return nil
+}
 
-	return
+func validatePortSelector(selector *routingv2.PortSelector) error {
+	if selector == nil {
+		return nil
+	}
+
+	if name := selector.GetName(); name != "" {
+		return validateSubsetName(name)
```

Comment:
`validateSubsetName` just calls IsDNS1123 and returns a formatted error because we validate subset names in several places.

Reply:
Nit: validate both conditions and return a combined error, rather than the first error we get.

Reply:
`PortSelector` is OneOf number or name, so we only want to validate one or the other. The logic is less clean than ideal because we cannot distinguish between an unset value and the zero value.